lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -765,6 +765,27 @@ defmodule ExUnit.Case do
     end
   end
 
+  @doc """
+  Returns the most recently registered test case as an `%ExUnit.Test{}`
+  struct.
+
+  This is used by third-party utilities to allow compile-time configuration
+  using test tags without having to explicitly pass the test context at
+  run-time. It is indented to be invoked in macros before the test module
+  is compiled.
+
+  Raises if called with a module that has already been compiled.
+  """
+  @doc since: "1.15.0"
+  @spec most_recent_registered_test(env) :: ExUnit.Test.t() | nil
+  def most_recent_registered_test(%{module: mod}) do
```

Comment:
We just need to bikeshed on the name now. I feel it should be called `get_last_registered_test` but I don't have an argument, it is just gut feeling (fwiw). Any thoughts @sabiwara @ericmj @whatyouhide?

Reply:
After sitting with it for another day, and having just implemented `Module.get_last_attribute`, I agree with you that `get_last_registered_test` is better. I'm going to change it over in a separate commit so it's easy to revert/update if other folks have a good alternative!

Reply:
`get_last_registered_test` sounds perfect, the naming is consistent with both [`register_test`](https://hexdocs.pm/ex_unit/ExUnit.Case.html#register_test/6) and `Module.get_last_attribute` 👌